pkg/yqlib/operator_filter_test.go

```diff
@@ -0,0 +1,49 @@
+package yqlib
+
+import (
+	"testing"
+)
+
+var filterOperatorScenarios = []expressionScenario{
+	{
+		description: "Filter array",
+		document:   `[1,2,3]`,
+		expression: `filter(. < 3)`,
+		expected: []string{
+			"D0, P[], (!!seq)::- 1\n",
+			"D0, P[], (!!seq)::- 2\n",
```

Comment:
this isn't actually the right result - this means the filter function is return two arrays - each with a single element. It should return a single array with two elements

```
"D0, P[], (!!seq)::- 1\n- 2\n",
```

I think if you refactor to use map and select under the hood - it would do the right thing.